packages/react/src/components/form/form.tsx

```diff
@@ -14,42 +14,127 @@ import {
   ConfigurationOverrides,
   ConfigurationOverridesProps,
 } from "../configuration-overrides";
-import { LoginOptions } from "../../domain/types";
+import { Handle, HandleType, LoginOptions } from "../../domain/types";
+import React, { useCallback } from "react";
+import { useSlots } from "../slot";
+import { Factor } from "@slashid/slashid";
+import { Flag } from "../input";
 
 export type Props = ConfigurationOverridesProps & {
   className?: string;
   onSuccess?: CreateFlowOptions["onSuccess"];
   middleware?: LoginOptions["middleware"];
+  children?: React.ReactNode;
 };
 
-export const Form: React.FC<Props> = ({
+type PayloadOptions = {
+  handleType?: HandleType;
+  handleValue?: string;
+  flag?: Flag;
+};
+
+export type InternalFormContextType = {
+  flowState: ReturnType<typeof useFlowState> | null;
+  lastHandle?: Handle;
+  submitPayloadRef: React.MutableRefObject<PayloadOptions>;
+  handleSubmit: (factor: Factor, handle?: Handle) => void;
+  selectedFactor?: Factor;
+  setSelectedFactor: React.Dispatch<React.SetStateAction<Factor | undefined>>;
+};
+export const InternalFormContext = React.createContext<InternalFormContextType>(
```

Comment:
Context is used to share values between nested components.

Reply:
Does this work with SSR? I guess only the initial state matters for SSR right?

Reply:
Didn't we drop the Context option because of SSR issues? @ikovic did you manage to find a workaround for that?